src/dex/uniswap-v3/contract-math/Oracle.ts

```diff
@@ -150,7 +150,8 @@ export class Oracle {
     }
 
     beforeOrAt = state.observations[(index + 1) % cardinality];
-    if (!beforeOrAt.initialized) beforeOrAt = state.observations[0];
+    if (!beforeOrAt || !beforeOrAt.initialized)
```

Comment:
The change here is the same as the one in 101: https://github.com/Uniswap/v3-core/blob/main/contracts/libraries/Oracle.sol#L223